modules/generator/registry/native_histogram.go

```diff
@@ -141,6 +141,8 @@ func (h *nativeHistogram) newSeries(labelValueCombo *LabelValueCombo, value floa
 			NativeHistogramBucketFactor:     1.1,
 			NativeHistogramMaxBucketNumber:  100,
 			NativeHistogramMinResetDuration: 15 * time.Minute,
+			// TODO enable examplars on native histograms
+			NativeHistogramMaxExemplars: -1,
```

Comment:
it does look like we try to write exemplars for native histograms. are you saying this code is not working?

Reply:
Yes, partly... I think there is something wrong with this code:
https://github.com/grafana/tempo/blob/2b00b4ece4a7d3b4779c38c9760d5eccc0bf322b/modules/generator/registry/native_histogram.go#L235-L250
(there is even a TODO that questions whatever is happening here!)

If I understand it correctly, at this point we have collected both the classic and the native histograms already.

If I enable `NativeHistogramMaxExemplars` the tests fail because we get more exemplars than before (this is expected since we enabled native histograms exemplars) but also some empty non-initialized exemplars (all fields are 0 or nil).
So I believe we need a bit more logic here to make sure we grab the right exemplars and filter out duplicate or empty exemplars.

I didn't want to figure all that out in this upgrade PR, so disable the feature for now which keeps behaviour consistent with before.

Reply:
I have created an issue for this: https://github.com/grafana/tempo/issues/4482
I've linked the prometheus PR in which they changed their implementation.